test/e2e/kubectl.go

```diff
@@ -56,49 +48,131 @@ var _ = Describe("kubectl", func() {
 		expectNoError(err)
 	})
 
-	It("should create and stop a replication controller", func() {
-		defer cleanup(nautilusPath)
-
-		By("creating a replication controller")
-		runKubectl("create", "-f", nautilusPath)
-		validateController(c, nautilusImage, 2)
+	Describe("update-demo", func() {
+		var (
+			updateDemoRoot = filepath.Join(testContext.repoRoot, "examples/update-demo/v1beta1")
+			nautilusPath   = filepath.Join(updateDemoRoot, "nautilus-rc.yaml")
+			kittenPath     = filepath.Join(updateDemoRoot, "kitten-rc.yaml")
+		)
+
+		It("should create and stop a replication controller", func() {
+			defer cleanup(nautilusPath, updateDemoSelector)
+
+			By("creating a replication controller")
+			runKubectl("create", "-f", nautilusPath)
+			validateController(c, nautilusImage, 2)
+		})
+
+		It("should scale a replication controller", func() {
+			defer cleanup(nautilusPath, updateDemoSelector)
+
+			By("creating a replication controller")
+			runKubectl("create", "-f", nautilusPath)
+			validateController(c, nautilusImage, 2)
+			By("scaling down the replication controller")
+			runKubectl("resize", "rc", "update-demo-nautilus", "--replicas=1")
+			validateController(c, nautilusImage, 1)
+			By("scaling up the replication controller")
+			runKubectl("resize", "rc", "update-demo-nautilus", "--replicas=2")
+			validateController(c, nautilusImage, 2)
+		})
+
+		It("should do a rolling update of a replication controller", func() {
+			// Cleanup all resources in case we fail somewhere in the middle
+			defer cleanup(updateDemoRoot, updateDemoSelector)
+
+			By("creating the initial replication controller")
+			runKubectl("create", "-f", nautilusPath)
+			validateController(c, nautilusImage, 2)
+			By("rollingupdate to new replication controller")
+			runKubectl("rollingupdate", "update-demo-nautilus", "--update-period=1s", "-f", kittenPath)
+			validateController(c, kittenImage, 2)
+		})
 	})
 
-	It("should scale a replication controller", func() {
-		defer cleanup(nautilusPath)
-
-		By("creating a replication controller")
-		runKubectl("create", "-f", nautilusPath)
-		validateController(c, nautilusImage, 2)
-		By("scaling down the replication controller")
-		runKubectl("resize", "rc", "update-demo-nautilus", "--replicas=1")
-		validateController(c, nautilusImage, 1)
-		By("scaling up the replication controller")
-		runKubectl("resize", "rc", "update-demo-nautilus", "--replicas=2")
-		validateController(c, nautilusImage, 2)
-	})
+	Describe("guestbook", func() {
+		var (
+			guestbookPath       = filepath.Join(testContext.repoRoot, "examples/guestbook")
+			frontendSelector    = "name=frontend"
```

Comment:
The vars here that are just strings should be defined in a const block instead of a var block:

const (
  frontendSelector = "name=frontend"
  redisMasterSelector = "name=redis-master"
  redisSlave = "name=redis-slave"
)
var guestbookPath = filepath.Join(...)